src/kernel.cr

```diff
@@ -1,13 +1,13 @@
 {% if flag?(:win32) %}
   STDIN  = IO::FileDescriptor.new(0)
-  STDOUT = (IO::FileDescriptor.new(1)).tap { |f| f.flush_on_newline = true }
-  STDERR = (IO::FileDescriptor.new(2)).tap { |f| f.flush_on_newline = true }
+  STDOUT = IO::FileDescriptor.new(1).tap { |f| f.flush_on_newline = true }
+  STDERR = IO::FileDescriptor.new(2).tap { |f| f.flush_on_newline = true }
 {% else %}
   require "c/unistd"
 
-  STDIN  = IO::FileDescriptor.new(0, blocking: LibC.isatty(0) == 0)
-  STDOUT = (IO::FileDescriptor.new(1, blocking: LibC.isatty(1) == 0)).tap { |f| f.flush_on_newline = true }
-  STDERR = (IO::FileDescriptor.new(2, blocking: LibC.isatty(2) == 0)).tap { |f| f.flush_on_newline = true }
+  STDIN  = IO::FileDescriptor.from_stdio(0)
+  STDOUT = IO::FileDescriptor.from_stdio(1).tap { |f| f.flush_on_newline = true }
+  STDERR = IO::FileDescriptor.from_stdio(2).tap { |f| f.flush_on_newline = true }
 {% end %}
 
```

Comment:
I would rather not use tap for `win32` and repeat for `unix` (`&.` can be used btw), but put at the end
```crystal
STDOUT.flush_on_newline = true
STDERR.flush_on_newline = true
```

Reply:
I don't know about const assignment in crystal, but is it potentially going to be run out-of-order if I do that?  For example, it looked like you could use STDIN in the `main` method, before code in kernel.cr runs. (I could be wrong)

Reply:
Didn't know this detail about out of order execution.
Still, a style detail, `|f| f.` can be replaced by `&.` :smile: 
```crystal
{% if flag?(:win32) %}
  STDOUT = IO::FileDescriptor.new(1).tap &.flush_on_newline = true
  STDERR = IO::FileDescriptor.new(2).tap &.flush_on_newline = true
{% else %}
  require "c/unistd"
  STDOUT = IO::FileDescriptor.from_stdio(1).tap &.flush_on_newline = true
  STDERR = IO::FileDescriptor.from_stdio(2).tap &.flush_on_newline = true
{% end %}
```